boards/shields/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: Apache-2.0
+
+foreach(shield_dir ${SHIELD_DIRS})
+  # To avoid a lot of empty CMakeLists.txt files we assume it is not an
+  # error if it is missing
+  if(EXISTS ${shield_dir}/CMakeLists.txt)
+    # Out-of-tree shield directories will not be a subdirectory,
+    # use the shield name as a relative cmake binary_dir
```

Comment:
this code is not using the shield name as cmake binary dir.
It's using the folder name where the shield is defined.

For example, if having:
```
board
 |-- shields
      |-- foo
           |-- bar.overlay
```
then the shield name is `bar`, but the value you're getting below with `cmake_path(GET shield_dir FILENAME shield)` is `foo`.

There doesn't have to be a one to one map of the shield dir to the shield name, so either fix the comment or use the shield name and fetch the dir using the `SHIELD_DIR_${shield_name}`.